builder/hcloud/builder_acc_test.go

```diff
@@ -35,7 +35,7 @@ func TestBuilderAcc_basic(t *testing.T) {
 const testBuilderAccBasic = `
 {
 	"builders": [{
-		"type": "test",
+		"type": "hcloud",
```

Comment:
So I saw you previously changed the check above to error if HCLOUD_TOKEN was not set. 
```go
	if v := os.Getenv("HCLOUD_TOKEN"); v == "" {
		t.Error("HCLOUD_TOKEN must be set for acceptance tests")
	}
```
But that failed because the check to determine if this is an acceptance and skipped is tucked inside the TestPlugin function call [here](https://github.com/hashicorp/packer-plugin-sdk/blob/4fb8273a56f0e5b27596bde9196f274242fe95b2/acctest/pluginacc.go#L64C1-L64C1) - this surprised me . 

Does moving the environment variable check into Setup work correctly? 

```go
	testCase := &acctest.PluginTestCase{
		Name: "hcloud_basic_test",
		Setup: func() error {
			if v := os.Getenv("HCLOUD_TOKEN"); v == "" {
				return fmt.Errorf("HCLOUD_TOKEN must be set for acceptance tests")
			}
			return nil
		},
		Template: testBuilderAccBasic,
		Check: func(buildCommand *exec.Cmd, logfile string) error {
			if buildCommand.ProcessState != nil {
				if buildCommand.ProcessState.ExitCode() != 0 {
					return fmt.Errorf("Bad exit code. Logfile: %s", logfile)
				}
			}
			return nil
		},
	}
	acctest.TestPlugin(t, testCase)
}

```

Reply:
Nice catch, using the setup does fix the separation between unit and acceptance tests. Thanks